### PR TITLE
ENHANCED: Added 'c_cxx' flag and env SWIPL_CXX for packs

### PR DESCRIPTION
**Pull request by kamahen**
I did a quick check of `buildenv.sh` when I did `pack_install(.)` for a pack and it appears to be correct -- `SWIPL_CXX='g++'` was added.

### CODE REVIEW

File: cmake/BuildType.cmake [hunk 1]
```diff
@@ -49,7 +49,7 @@ if(CMAKE_COMPILER_IS_GNUCC)
     set(GCC_OPTFLAGS -O2)
   endif()
 
-  set(CMAKE_C_FLAGS_DEBUG "-DO_DEBUG -DO_DEBUG_ATOMGC -O0 ${GCC_GFLAGS} $ENV{CFLAGS}"
+  set(CMAKE_C_FLAGS_DEBUG "-DO_DEBUG -DO_DEBUG_ATOMGC -O0 ${GCC_GFLAGS} $ENV{CFLAGS} -DO_CHECK_TERM_REFS -DSECURE_GC -DCHK_SECURE -DO_MAINTENANCE -gdwarf-2 -g3"
```
Comment (kamahen) [inline cmake/BuildType.cmake:52]: You might not agree with this change -- but it has the flags that caused https://github.com/SWI-Prolog/swipl-devel/issues/1292 when doing a debug build.

File: library/build/tools.pl [hunk 2]
```diff
@@ -435,12 +442,25 @@
...
+    has_program(path(CXX), _),
+    !.
+
 preferred_c_compiler(CC) :-
     current_prolog_flag(c_cc, CC).
 preferred_c_compiler(gcc).
 preferred_c_compiler(clang).
 preferred_c_compiler(cc).
 
+preferred_cxx_compiler('g++').
```
Comment (kamahen) [inline library/build/tools.pl:459]: Add:
```prolog
preferred_c_compiler(CXX) :-
    current_prolog_flag(c_cxx, CXX).
```

File: library/build/tools.pl [hunk 3]
```diff
@@ -435,12 +442,25 @@
...
+    getenv('CXX', CXX),
+    !.
+default_cxx_compiler(CXX) :-
+    preferred_cxx_compiler(CXX),
+    has_program(path(CXX), _),
+    !.
+
 preferred_c_compiler(CC) :-
     current_prolog_flag(c_cc, CC).
 preferred_c_compiler(gcc).
```
Comment (kamahen) [inline library/build/tools.pl:455]: See https://github.com/SWI-Prolog/swipl-devel/issues/1294

Changing this to `preferred_c_compiler(path(gcc))` breaks `build_step(configure, ...)`.

**Review by kamahen — COMMENTED** [review]
This PR needs more work - See https://github.com/SWI-Prolog/swipl-devel/issues/1294

File: cmake/BuildType.cmake (hunk 1 above)
Comment (JanWielemaker) [reply cmake/BuildType.cmake:52]: Skipped this :smile:

File: library/build/tools.pl [hunk 4]
```diff
@@ -431,15 +438,30 @@
...
+    preferred_cxx_compiler(CXX0),
+    has_program(CXX0, CXX),
     !.
 
 preferred_c_compiler(CC) :-
     current_prolog_flag(c_cc, CC).
-preferred_c_compiler(gcc).
-preferred_c_compiler(clang).
-preferred_c_compiler(cc).
+preferred_c_compiler(path(gcc)).
```
Comment (JanWielemaker) [inline library/build/tools.pl:455]: I could find no good reason for the path() wrapper.   has_program/2 adds this if necessary.   If this fixes some issue I think it is has_program/2 that needs fixing.    So, I skipped this.

**Comment by JanWielemaker** [comment]
Applied after the mentioned changes and some documentation changes.

File: library/build/tools.pl (hunk 4 above)
Comment (kamahen) [reply library/build/tools.pl:455]: Line 521 (program_install_prefix/1) has `has_program(path(Name), EXE),` ... is it wrong?

`has_program/2` uses `pack_build_path(...)`, not `path(...)` ... presumably the former is better?

**Pull request closed without merging**